react-components/src/hooks/useMappedEquipmentBy3DModelsList.tsx

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { type UseQueryResult, useQuery } from '@tanstack/react-query';
+import { type FdmAssetMappingsConfig } from '..';
+import { useFdmSdk } from '../components/RevealContainer/SDKProvider';
+
+export const useMappedEquipmentBy3DModelsList = (
+  Default3DFdmConfig: FdmAssetMappingsConfig,
+  modelsList: Array<{ modelId: number; revisionId: number }> = []
+): UseQueryResult<string[]> => {
+  const fdmClient = useFdmSdk();
+
+  return useQuery(
+    ['reveal', 'react-components', ...modelsList],
```

Comment:
It is a little akward deconstructing the model list here, since you could just replace an object with the same modelId and revisionId and it would cache miss. But I don't really know a better way sooo... 🤷‍♂️

Reply:
Changed the query key since when destructuring it also included transformation and all other stuff about the model. After reading docs on keys I suspect converting the array to `modelId`s list works quite good. It checks for equality of elements, not actual instances of elements of the query key array. See [link](https://tanstack.com/query/v4/docs/react/guides/query-keys)